src/hoa-rotator.js

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+
+/**
+ * @fileOverview Sound field rotator for higher-order-ambisonics decoding.
```

Comment:
It seems like many small functions in this class are not supposed to be exposed to the outside. Factoring them out as non-class-member would be great. Or perhaps we should think about creating a library file with them.